src/server/JasmineGraphInstanceFileTransferService.cpp

```diff
@@ -24,26 +25,30 @@ void *filetransferservicesession(void *dummyPt) {
     filetransferservicesessionargs *sessionargs = (filetransferservicesessionargs *)dummyPt;
     int connFd = sessionargs->connFd;
     delete sessionargs;
-    char data[301];
-    bzero(data, 301);
-    read(connFd, data, 300);
-    string fileName = (data);
+    char data[INSTANCE_DATA_LENGTH + 1];
+    string fileName = Utils::read_str_wrapper(connFd, data, INSTANCE_DATA_LENGTH);
     string filePathWithName =
         Utils::getJasmineGraphProperty("org.jasminegraph.server.instance.datafolder") + "/" + fileName;
 
-    write(connFd, JasmineGraphInstanceProtocol::SEND_FILE.c_str(), JasmineGraphInstanceProtocol::SEND_FILE.size());
-    int bytesReceived = 0;
-    char buffer[1024];
+    Utils::send_str_wrapper(connFd, JasmineGraphInstanceProtocol::SEND_FILE_LEN);
+    string fsizeStr = Utils::read_str_wrapper(connFd, data, INSTANCE_DATA_LENGTH);
+    int fsize = stoi(fsizeStr);
+    Utils::send_str_wrapper(connFd, JasmineGraphInstanceProtocol::SEND_FILE);
+    char buffer[4096];
```

Comment:
Why do we use a buffer of 4096 here? Can we move 4096 to a meaningful constant declaration?

Reply:
It's the buffer size, and it's not used anywhere else in this file.

Reply:
Ok